doc/remailers.rst

```diff
@@ -0,0 +1,125 @@
+the remailer is honest at the time of sending.
+
+This document focuses on seamless support for re-encryption based end-to-end
+security in existing remailers, in keeping with the more general Autocrypt
+approach.
+
+Autocrypt-enabled Remailer
+----
+
+An Autocrypt-enabled remailer maintains an Autocrypt key of its own. During
+regular operation, it collects Autocrypt-headers of all list participants via
+incoming mail, to keep track of their keys. Then, depending on the availability
+of encryption keys for participants, the remailer injects its own
+Autocrypt-headers into outgoing mail to signal availability of encryption.
+
+More specifically, an Autocrypt-enabled mailing list adds three behaviors to its
+remailing logic:
+
+1) For all incoming messages, keep track of Autocrypt-headers. This mechanism is
+   the same that is used by clients, and requires no modifications.
+2) Inject an Autocrypt-header into outgoing messages. This header serves the
+   same purpose as usual, but adds some details for the mailing list use case.
+   The header might only be injected under certain conditions, e.g. if
+   encryption keys are available for a certain number of subscribers.
+3) When receiving an encrypted message, decrypt it and re-encrypt for individual
+   delivery to each Autocrypt-enabled subscriber. Subscribers for whom no keys
+   are available do not receive the message, but optionally a notification of
+   this circumstance (see below).
+
+The presented mechanism works well for relatively small groups of mostly active
+and Autocrypt-enabled subscribers, but becomes increasingly ineffective with
+larger numbers of passive or non Autocrypt-enabled subscribers. As with any
+encrypted group communication, the benefits of end-to-end encryption also dilute
+with larger numbers of recipients.
+
+For compatibility with clients that are not Autocrypt-capable, the remailer
+software might want to offer manual configuration of keys, e.g. via a web
+interface.
+
+Injecting Reamiler Autocrypt-Headers
+~~~~~
+
+To inform subscribers of its key, the remailer injects an Autocrypt-header into
+outgoing messages. The header format is the same as usual, where the ``addr``
+attribute is the address of the list, and ``keydata`` is their Autocrypt key.
+
+In addition to those two attributes, the remailer SHOULD add the
+``_list-recommendation`` attribute, and MAY add ``_list-recipients`` attribute.
+Both of these attributes are exclusive for use by remailers, and MUST NOT be
+used by regular mailbox addresses. Both of these attributes are non-critical
+(starting with an underscore), which will cause an Autocrypt-capable client that
+does not handle these attributes to skip over them, falling back to the usual
+Autocrypt behavior.
+
+For example::
+
+    Autocrypt: addr=post@list.example; _list-recommendation=mandatory; keydata=XXX
+    Autocrypt: addr=post@list.example; _list-recommendation=available; _list-recipients=7/15; keydata=XXX
+
```

Comment:
These examples should be separated.  without separation, it might imply that the same header is being set twice in a single message.